src/Bicep.Core/TypeSystem/Radius/V3/KnownComponents.cs

```diff
@@ -81,6 +81,39 @@ public static ComponentData MakeContainer()
                 functions: null);
             var portsProperty = new TypeProperty("ports", portsType, TypePropertyFlags.None);
 
+            var ephemeralVolume = new ObjectType(
+                name: "ephemeral",
+                validationFlags: TypeSymbolValidationFlags.Default,
+                properties: new TypeProperty[] {
+                    new TypeProperty("kind", new StringLiteralType("ephemeral"), TypePropertyFlags.Required),
+                    new TypeProperty("mountPath", LanguageConstants.String, TypePropertyFlags.Required),
+                    new TypeProperty("managedStore", UnionType.Create(new StringLiteralType("memory"), new StringLiteralType("disk")), TypePropertyFlags.None),
+                },
+                additionalPropertiesType: null,
+                additionalPropertiesFlags: TypePropertyFlags.None,
+                functions: null);
+
+            var persistentVolume = new ObjectType(
+                name: "persistent",
+                validationFlags: TypeSymbolValidationFlags.Default,
+                properties: new TypeProperty[] {
+                    new TypeProperty("kind", new StringLiteralType("persistent"), TypePropertyFlags.Required),
+                    new TypeProperty("mountPath", LanguageConstants.String, TypePropertyFlags.Required),
+                    new TypeProperty("source", LanguageConstants.String, TypePropertyFlags.Required),
+                    new TypeProperty("rbac", UnionType.Create(new StringLiteralType("read"), new StringLiteralType("write")), TypePropertyFlags.None),
+                },
+                additionalPropertiesType: null,
+                additionalPropertiesFlags: TypePropertyFlags.None,
+                functions: null);
+
+            var volumesType = new DiscriminatedObjectType(
+                name: "volumes",
+                validationFlags: TypeSymbolValidationFlags.Default,
+                discriminatorKey: "kind",
+                unionMembers: new ITypeReference[]{ephemeralVolume, persistentVolume}
+                );
+            var volumesProperty = new TypeProperty("volumes", volumesType, TypePropertyFlags.None);
```

Comment:
Can you add descriptions for the properties? It's ok to copy-paste from the schema

Reply:
`volumesProperty` isn't quite right. We wanted to support a map-like of volumes.

What's here is:

```
volumes: {
  kind: 'ephemeral'
  ...
}
```

We want:

```
volumes: {
  foo: {
      kind: 'ephemeral'
    ...
  }
}
```

To do this you need to define a type for `volume` and a type for `volumes` (w/ Additionalproperties -> `volume`)